### PR TITLE
Client-side encryption metadata deserialization case-insensitive

### LINKED ISSUE #29750: [BUG] NullReferenceException reading V1.0 camel cased client-side encryption metadata
### Library name and version

Azure.Storage.Blobs 12.13.0

### Describe the bug

NullReferenceException when attempting to deserialize client side encryption metadata (encryptiondata) that was serialized with a previous version of the SDK configure to use camel casing (stack below).  The problem appears to be in the various EncryptionDataSerializer.ReadPropertyValue methods that use JsonProperty.NameEquals to do a case-sensitive name comparison.

"innermostType": System.NullReferenceException,
"innermostMessage": Object reference not set to an instance of an object.,
"details":    at Azure.Storage.Blobs.BlobClientSideDecryptor.GetAndValidateEncryptionDataOrDefault(IDictionary`2 metadata)
   at Azure.Storage.Blobs.Specialized.BlobBaseClient.<OpenReadInternal>d__93.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Storage.Blobs.Specialized.BlobBaseClient.<OpenReadAsync>d__91.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.TemporaryAttachmentStore.<>c__DisplayClass23_0.<<GetAttachmentContentAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.TemporaryAttachmentStore.<AzureBlobOperationAsync>d__33`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.TemporaryAttachmentStore.<GetAttachmentContentAsync>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Bot.ChannelConnector.AttachmentsController.<>c__DisplayClass3_0.<<GetAttachment>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Microsoft.Bot.ChannelConnector.WebApiControllerExtensions.<ExecuteAndRespond>d__0`1.MoveNext(),

### Expected behavior

Json deserialization should be case-insensitive to support reading existing metadata

### Actual behavior

NullReferenceException when attempting to read existing blobs that use client-side encryption and were written by a previous version of the SDK

### Reproduction Steps

1) set the default JsonSerializer to camelCase:
   JsonConvert.DefaultSettings = () => new JsonSerializerSettings()
   {
     ContractResolver = new CamelCasePropertyNamesContractResolver()
   };
2) Write a blob with client-side encryption enabled using V1 encryption with the previous SDK (Microsoft.Azure.Storage.Blob V11.2.3)
3) Verify that the encryption metadata Json is camelCased.  It should look like this:
{"encryptionMode":"FullBlob","wrappedContentKey":{"keyId":"key1","encryptedKey":"B79XoZ7Ky+/9xJps7SFFLZOedlSfuLP+mukLtuUKEpWNZHUR5CIbsQ==","algorithm":"A256KW"},"encryptionAgent":{"protocol":"1.0","encryptionAlgorithm":"AES_CBC_256"},"contentEncryptionIV":"/e8I2sr5ow1jtMw/oDJB3g==","keyWrappingMetadata":{"encryptionLibrary":".NET 11.2.3"}}
4) Attempt to read the blob with the new SDK (Azure.Storage.Blobs V12.13.0)

Another more simple repro is to write a blob using the 12.13.0 SDK and manually change the casing of the encryptionData to be camelCased and then attempt to read back the blob. 

### Environment

_No response_

### PR DESCRIPTION
**Pull request by jaschrep-msft**
Resolves #29750

Queues untested. Nature of issue only leaves persisted blob encryption metadata in a state that needs accommodation after the fact.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## 12.14.0-beta.1 (Unreleased)
 - Added support for service version 2021-10-04.
 - Added support for SDK-calculated transactional checksums on data transfer.
+- Added support for downloading blobs with bugged client-side encryption metadata from previous library verions.
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Blobs/CHANGELOG.md:6]: NIT - spelling "verions" to "ver**s**ion"

File: sdk/storage/Azure.Storage.Queues/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Release History
 
 ## 12.12.0-beta.1 (Unreleased)
-- This release contains bug fixes to improve quality.
+- Added support for receiving queue messages with bugged client-side encryption metadata from previous library verions.
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Queues/CHANGELOG.md:4]: same, spelling "verions" to "ver**s**ions"

**Review by amnguye — APPROVED** [review]

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**